cmake/modules/SwiftSource.cmake

```diff
@@ -78,6 +78,17 @@ function(handle_swift_sources
       list(APPEND swift_compile_flags "-Xcc" "-D__SWIFT_CURRENT_DYLIB=swiftCore")
     endif()
 
+    precondition(NOT SWIFTSOURCES_IS_STDLIB OR NOT SWIFTSOURCES_MODULE_DIR,
+      "Cannot provide MODULE_DIR if IS_STDLIB is set")
+
+    if(SWIFTSOURCES_IS_STDLIB)
```

Comment:
Okay, so I tried this for a while -- making `TARGET_LIBRARY` infer `MODULE_DIR` instead of `IS_STDLIB`. CMake, in turn, stopped building `SwiftSyntax` entirely. Then I moved it from `tools/` to `stdlib/public/`, and it built just fine.

There seems to be some more implicit behavior (maybe because `tools/` is included before `stdlib/`?) that's preventing this.